core/services/database/TableAnalysis.php

```diff
@@ -60,63 +63,22 @@ public function tableIsEmpty($table_name)
      * Indicates whether or not the table exists. $table_name can optionally
      * have the $wpdb->prefix on the beginning, or not.
      *
-     * @global \wpdb $wpdb
-     * @global array EZSQL_Error
-     * @param        $table_name
+     * @global wpdb $wpdb
+     * @param string $table_name
      * @return bool
      */
     public function tableExists($table_name)
     {
-        global $wpdb, $EZSQL_ERROR;
+        global $wpdb;
         $table_name = $this->ensureTableNameHasPrefix($table_name);
-        // ignore if this causes an sql error
-        $old_error = $wpdb->last_error;
-        $old_suppress_errors = $wpdb->suppress_errors();
-        $old_show_errors_value = $wpdb->show_errors(false);
-        $ezsql_error_cache = $EZSQL_ERROR;
-        $wpdb->get_results("SELECT * from $table_name LIMIT 1");
-        $wpdb->show_errors($old_show_errors_value);
-        $wpdb->suppress_errors($old_suppress_errors);
-        $new_error = $wpdb->last_error;
-        $wpdb->last_error = $old_error;
-        $EZSQL_ERROR = $ezsql_error_cache;
-        // if there was a table doesn't exist error
-        if (! empty($new_error)) {
-            if (in_array(
-                \EEH_Activation::last_wpdb_error_code(),
-                array(
-                    1051, // bad table
-                    1109, // unknown table
-                    117, // no such table
-                )
-            )
-                ||
-                preg_match(
-                    '~^Table .* doesn\'t exist~',
-                    $new_error
-                ) // in case not using mysql and error codes aren't reliable, just check for this error string
-            ) {
-                return false;
-            } else {
-                // log this because that's weird. Just use the normal PHP error log
-                error_log(
-                    sprintf(
-                        __(
-                            'Event Espresso error detected when checking if table existed: %1$s (it wasn\'t just that the table didn\'t exist either)',
-                            'event_espresso'
-                        ),
-                        $new_error
-                    )
-                );
-            }
-        }
-        return true;
+        $query = $wpdb->prepare( 'SHOW TABLES LIKE %s', $wpdb->esc_like( $table_name ) );
+        return $wpdb->get_var( $query ) === $table_name;
```

Comment:
Most likely phpcs won't like this.

```suggestion
        $query = $wpdb->prepare('SHOW TABLES LIKE %s', $wpdb->esc_like($table_name));
        return $wpdb->get_var($query) === $table_name;
```

Reply:
yup saw that thnx...
this change also opened a can of worms with the unit tests because they were all set up to expect errors to be thrown during activation, so I'm currently trying to fix those issues right now